editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -823,6 +828,45 @@ export const sortBreadcrumbs = (
       );
 };
 
+const handleSetValue = (
+  flow: Store.flow,
+  id: string,
+  acc: Record<string, any>,
+  responseData: Record<string, any> | undefined,
+  passport: Store.passport,
+): Store.passport => {
+  const { operation, fn } = flow[id]?.data as SetValue;
+  let previousValues = acc.data?.[fn];
+
+  // We do not amend values set at objects
+  // These are internal exceptions we do not want to allow users to edit
+  // e.g. property.boundary.title
+  const isObject =
+    typeof previousValues === "object" &&
+    !Array.isArray(previousValues) &&
+    previousValues !== null;
+  if (isObject) return passport;
+
+  previousValues = formatPreviousValues(previousValues);
+  const currentValue = responseData?.[fn] || [];
+
+  if (operation === "remove") {
```

Comment:
Unfortunately I haven't been able to replicate the slightly odd behaviour I was seeing in a clear simple flow. Which potentially suggests it's too much of an edge case to worry about. 

On the pizza I've updated [FOIYNPP](https://2914.planx.pizza/lambeth/find-out-if-you-need-planning-permission) to add some extra `setValue` nodes and dummy questions to act as moments to run `computePassport()` and assess the values. 

What I think I'm seeing is that `findOut` isn't being removed from `application.type` and after the `append` later on it's appearing as the second item in the array. 

I feel like I must be missing something 🤷

Reply:
I've not been able to replicate this, but will set up a series of test flows to manual testing.